buildSrc/src/main/java/org/elasticsearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -0,0 +1,258 @@
+        if (unreleasedInfo != null) {
+            assert distribution.getBundledJdk();
+            return projectDependency(project, unreleasedInfo.gradleProjectPath, distributionProjectName(distribution));
+        }
+
+        if (distribution.getType().equals("integ-test-zip")) {
+            return "org.elasticsearch.distribution.integ-test-zip:elasticsearch:" + distribution.getVersion();
+        }
+
+        String extension = distribution.getType();
+        String classifier = "x86_64";
+        if (distribution.getType().equals("archive")) {
+            extension = distribution.getPlatform().equals("windows") ? "zip" : "tar.gz";
+            classifier = distribution.getPlatform() + "-" + classifier;
+        }
+        return FAKE_GROUP + ":elasticsearch" + (distribution.getType().equals("oss") ? "-oss:" : ":")
+            + distribution.getVersion() + ":" + classifier + "@" + extension;
+    }
+
+    private static Dependency projectDependency(Project project, String projectPath, String projectConfig) {
+        if (project.findProject(projectPath) == null) {
+            throw new GradleException("no project [" + projectPath + "], project names: " + project.getRootProject().getAllprojects());
+        }
+        Map<String, Object> depConfig = new HashMap<>();
+        depConfig.put("path", projectPath);
+        depConfig.put("configuration", projectConfig);
+        return project.getDependencies().project(depConfig);
+    }
+
+    private static String distributionProjectPath(ElasticsearchDistribution distribution) {
+        String projectPath = ":distribution";
+        if (distribution.getType().equals("integ-test-zip")) {
+            projectPath += ":archives:integ-test-zip";
+        } else {
+            projectPath += distribution.getType().equals("archive") ? ":archives:" : ":packages:";
+            projectPath += distributionProjectName(distribution);
+        }
+        return projectPath;
+    }
+
+    private static String distributionProjectName(ElasticsearchDistribution distribution) {
+        String projectName = "";
+        if (distribution.getFlavor().equals("oss")) {
+            projectName += "oss-";
+        }
+        if (distribution.getBundledJdk() == false) {
+            projectName += "no-jdk-";
+        }
+        if (distribution.getType().equals("archive")) {
+            String platform = distribution.getPlatform();
+            projectName += platform + (platform.equals("windows") ? "-zip" : "-tar");
+        } else {
+            projectName += distribution.getType();
+        }
+        return projectName;
+    }
+
+    private static String configName(String prefix, ElasticsearchDistribution distribution) {
+        return prefix + "_" + distribution.getVersion() + "_" + distribution.getPlatform() + "_" + distribution.getType() + "_"
+            + distribution.getFlavor() + (distribution.getBundledJdk() ? "" : "_nojdk");
+    }
+
+    private static String capitalize(String s) {
```

Comment:
I'd like to leave that as a future cleanup. I'm not against it, but would like to minimize expanding the scope of this PR more, as I would want such a utils class to have tests.

Reply:
That's fine, but I'd really like us to become more open to these kinds of minor refactorings in the course of PRs. There are countless "we can do this in a separate PR" comments that are never followed up on because they are small and low priority. In practice, I fear this stuff just never gets done. I'd prefer untested DRY code to untested duplicate code.